### PR TITLE
Check that landsat scenes are in same projection

### PR DESCRIPTION
**Pull request by jacquelynsmale**

### CODE REVIEW

**Comment by jacquelynsmale** [comment]
I've got the current version downloading a copy of the scene to be re-projected, which was the only way I could get gdal.Warp to work. 

Now, I'm facing an error that `Exception: Upper bound of coregistered image index should be <= size of image1 (and image2) minus 1`. @jhkennedy , @forrestfwilliams have you come across this error before?

**Comment by jhkennedy** [comment]
Okay, I got this working for a L5 pair on my laptop. Open questions I'd like @jacquelynsmale and @forrestfwilliams thoughts on:
* Should we run the FFT filter before reporjecting? I'm leaning yes.
* Similarly, should we run the Wallis no-data fill filter before reprojecting? I'm also leaning yes though that will be a bigger lift.

**Comment by jacquelynsmale** [comment]
@jhkennedy @forrestfwilliams I would also lean towards filtering then reprojecting.

File: environment.yml [hunk 1]
```diff
@@ -30,6 +30,6 @@ dependencies:
   - boto3
   - matplotlib-base
   - netCDF4
-  - numpy
+  - numpy<1.24
```
Comment (jhkennedy) [inline environment.yml:33]:
```suggestion
  - numpy<1.24  # https://github.com/isce-framework/isce2/pull/639
```

**Comment by jhkennedy** [comment]
Alright, I've confirm we've got the filtering for L7 working (as in, the filtered TIFs look correct), but we're still getting completely zero output products and browse images.

Looking out the outputs:
* All the `window*.tif` outputs from Geogrid look correct as compared to `develop`
* in `autoRIFT_intermediate.nc`, `SeachLimitX`, `SearchLimitY` and `noDataMask` look correct as compared to `develop`
* in `autoRIFT_intermediate.nc`, `Dx`, `Dy`, `InterpMask`, `ChipSizeX` are all zero, leading to :point_down: 
* The `offset.tif`, and therefor `velocity.tif`, from autoRIFT are all zero

Next place to look at is how `Dx`, `Dy`, `InterpMask`, `ChipSizeX` are set.

**Comment by jhkennedy** [comment]
Okay, I suspect it's something going on with [`uniform_data_type`](https://github.com/nasa-jpl/autoRIFT/blob/master/geo_autoRIFT/autoRIFT/autoRIFT.py#L345) which is called in `vend/testautoRIFT.py` right after the pre-processing steps:
* Preprocessing called in [lines 286-323](https://github.com/ASFHyP3/hyp3-autorift/pull/173/files#diff-aec97127e1a0f9c669793fef4650bdfcd75ea2522b8826a4df963ef623d937a7R286-R323)
* uniform data type called fight after on [line 327 ](https://github.com/ASFHyP3/hyp3-autorift/pull/173/files#diff-aec97127e1a0f9c669793fef4650bdfcd75ea2522b8826a4df963ef623d937a7R327)

switching `obj.DataType` to `1` instead of the default `0` here:
https://github.com/ASFHyP3/hyp3-autorift/pull/173/files#diff-aec97127e1a0f9c669793fef4650bdfcd75ea2522b8826a4df963ef623d937a7R326

makes it so we output *some* data, albeit bogus data:
LE07_L1TP_063018_20040810_20200915_02_T1_X_LE07_L1TP_063018_20040911_20200915_02_T1_G0120V02_P070_IL_ASF_OD

**Comment by jhkennedy** [comment]
Along a different path, the highpass filter (HPS) is applied to all missions, so:
* Sentinel-2 would also need to be filtered before geogrid/autoRIFT
* Sentinel-1 would also need to be filtered before geogrid/autoRIFT, but *after* all the ISCE2 shenanigans
but we currently don't handle either of these cases (and removed the filter from autoRIFT workflow!).

We want to filter before reprojecting:
* for L45 because the filter relies on the shape of the valid input data
* for L7 because the no-data bands we fill get all kinds of funky when re-projecting

For L8/9, I don't think reprojecting will have much effect on the low-frequency features, and we likely don't *need* to filter ahead of time. For S2, and S1, we won't ever be re-projecting, so there's no need to filter ahead of time.

Therefore, I will walk back moving the high-pass filter before geogrid/autoRIFT for all missions.

**Comment by jhkennedy** [comment]
Back looking at these two scenes: 
`LE07_L1TP_063018_20040911_20200915_02_T1 LE07_L1TP_063018_20040810_20200915_02_T1`

With the changes I just pushed, I can now get output data that looks reasonable:
LE07_L1TP_063018_20040810_20200915_02_T1_X_LE07_L1TP_063018_20040911_20200915_02_T1_G0120V02_P023_IL_ASF_OD

But still not as good as from `develop`:
LE07_L1TP_063018_20040810_20200915_02_T1_X_LE07_L1TP_063018_20040911_20200915_02_T1_G0120V02_P038_IL_ASF_OD

**Comment by forrestfwilliams** [comment]
With the complexities you mentioned @jhkennedy, I'd also favor walking back the changes to the high-pass filter. Thanks!

**Comment by forrestfwilliams** [comment]
It looks like we haven't correctly implemented filtering for at least Landsat 7. Post uniform_data_type images created using `develop` clearly look Wallis filtered, but the images created using `landsat_reproject` don't.

**develop:**
zoomin_dev
**reproject:**
zoomin_reproject

**Comment by forrestfwilliams** [comment]
I've confirmed that the problems are also present pre-`uniform_data_type`. The same visual pattern as above is present, but the `develop` branch data is scaled -5 to 5 (correct for Wallis filter) and the `reproject` branch data is scaled 0 to 255 (incorrect for Wallis filter).

**Comment by jhkennedy** [comment]
@forrestfwilliams I think the wallis filter itself is working correctly. 

For the current HEAD of this branch, if you throw the TIFFs from the `filtered` directory into QGIS, you'll get:
image

which looks Wallis filtered to me, and has a range from approx. -5 to 5. `gdalinfo -stats` on those confirm they are float32 data type and have a range from approx. -5 to 5.

To get the output I do see, I had to uncomment these two lines:
https://github.com/ASFHyP3/hyp3-autorift/blob/landsat_reproject/hyp3_autorift/vend/testautoRIFT.py#L156-L157

If they are commented out, we get blank browse images and empty netCDF files again.

So, I think what's happing is that autorift at that point assumes the images are scaled from 0-255 (the landsat input images are!). The cast to `uint8` causes the scale to go from 0-255 because anything < 0 ends up wrapping to 255-X.

I'm having a hard time finding *where* these assumptions *are*, and really understanding what `uniform_data_type` is even trying to do to the uint8 images.

**Comment by jhkennedy** [comment]
But that begs the question, why did the L4/5 filter work? Isn't that also producing filtered images that range from -5 to 5??

**Comment by forrestfwilliams** [comment]
@jhkennedy I agree that the issue is likely occurring inside of `testAutoRIFT.py` after images are pre-filtered. The images in the filtered directory also look good to me. I'll work on tracking down this issue.

**Comment by forrestfwilliams** [comment]
It looks like we hadn't correctly set the path for the zeroMask files we create during pre-filtering, so they weren't being read in. This caused cascading errors. To get decent out put I've commented [these lines](https://github.com/ASFHyP3/hyp3-autorift/blob/landsat_reproject/hyp3_autorift/vend/testautoRIFT.py#L156-L157). They are the source of some issues, but I'll pick this back up later.

File: hyp3_autorift/vend/testautoRIFT.py [hunk 2]
```diff
@@ -539,6 +550,17 @@ def generateAutoriftProduct(indir_m, indir_s, grid_location, init_offset, search
...
+        
+        zero_mask = None
+        indir_m_zero = f'{indir_m.split(".")[0]}_zeroMask.{indir_m.split(".")[1]}'
+        indir_s_zero = f'{indir_s.split(".")[0]}_zeroMask.{indir_s.split(".")[1]}'
+        if os.path.exists(indir_m_zero) or os.path.exists(indir_s_zero):
+            m_zero, s_zero = loadProductOptical(indir_m_zero, indir_s_zero)
+            m_zero = m_zero.astype(np.uint8)
+            s_zero = s_zero.astype(np.uint8)
+
+            # FIXME: Or? Wallis uses "or" here, while wallis_fill uses "and" here.
```
Comment (forrestfwilliams) [inline hyp3_autorift/vend/testautoRIFT.py:562]: Also, I think `and` vs `or` is important here. Could be why we're getting slightly less data than `develop currently.

Comment (forrestfwilliams) [reply hyp3_autorift/vend/testautoRIFT.py]: I've decided to switch to `and` for now.

**Comment by forrestfwilliams** [comment]
Current version works for L7, but with a slight decrease in the number of valid pixels. Not sure where this difference is coming from.

**Comment by forrestfwilliams** [comment]
While there are some small changes, I think the `L7` test case is now functionally equivalent:
**Develop offset.tif (full output [here](https://ffwilliams2-shenanigans.s3.us-west-2.amazonaws.com/landsat-reproject/develop_l7.zip))**
develop
**Landsat_reproject offset.tif (full output [here](https://ffwilliams2-shenanigans.s3.us-west-2.amazonaws.com/landsat-reproject/feature_l7.zip))**
feature

**Comment by jhkennedy** [comment]
Alright! Looks like most everything is good to go for golden testing. @forrestfwilliams if you're happy with #198 , feel free to merge it **and** this one to develop.

**Comment by forrestfwilliams** [comment]
Took a look at a test image pair that needed correction. We're producing decent data in comparison to the V1 datasets:
**V1 vX Dataset**
Screen Shot 2023-01-04 at 8 42 03 AM
**New reprojected vX Dataset (data [here](https://ffwilliams2-shenanigans.s3.us-west-2.amazonaws.com/landsat-reproject/reprojected_l7.zip))
Screen Shot 2023-01-04 at 8 41 48 AM

**Comment by jhkennedy** [comment]
@forrestfwilliams I think we're just going to ignore it and it'll clean itself up next release

**Review by jhkennedy — APPROVED** [review]
:shipit:

**Pull request merged**